form/form.go

```diff
@@ -335,12 +341,17 @@ func uintEncoder(values *Values, v reflect.Value, keyParts []string, encodeZero
 	values.Add(FormatKey(keyParts), strconv.FormatUint(val, 10))
 }
 
-func reflectValue(values *Values, v reflect.Value, _ bool, keyParts []string) {
+// reflectValue is roughly the shared entry point of any AppendTo functions.
+// It's also called recursively in cases where a precise type isn't yet known
+// and its encoding needs to be deferred down the chain; for example, when
+// encoding interface{} or the values in an array or map containing
+// interface{}.
+func reflectValue(values *Values, v reflect.Value, encodeZero bool, keyParts []string) {
```

Comment:
I found myself forgetting the details of this implementation already, so tried to add a few more explanatory comments.